server/jetstream_benchmark_test.go

```diff
@@ -1174,3 +1174,133 @@ func BenchmarkJetStreamKV(b *testing.B) {
 		)
 	}
 }
+
+func BenchmarkJetStreamObjStore(b *testing.B) {
+	const (
+		verbose            = false
+		objStoreNamePrefix = "B_"
+		keyPrefix          = "K_"
+		seed               = 12345
+		init               = true
+	)
+
+	benchmarksCases := []struct {
+		clusterSize int
+		replicas    int
+		numKeys     int
+		minBytes    int
+		maxBytes    int
+		rwRatio     float64
+	}{
+		{3, 3, 1000, 1024, 102400, 1.0},    // 3 nodes, 1000 keys, 10KB-100KB values, reads only
+		{3, 3, 1000, 1024, 102400, 0.0},    // 3 nodes, 1000 keys, 10KB-100KB values, writes only
+		{3, 3, 1000, 1024, 102400, 0.5},    // 3 nodes, 1000 keys, 10KB-100KB values, 50/50 read/writes
+		{3, 3, 1000, 1024, 102400, 0.8},    // 3 nodes, 1000 keys, 10KB-100KB values, 80/20 read/writes
+		{3, 3, 1000, 102400, 1048576, 1.0}, // 3 nodes, 1000 keys, 100KB-1MB values, reads only
+		{3, 3, 1000, 102400, 1048576, 0.0}, // 3 nodes, 1000 keys, 100KB-1MB values, writes only
+		{3, 3, 1000, 102400, 1048576, 0.5}, // 3 nodes, 1000 keys, 100KB-1MB values, 50/50 read/writes
+		{3, 3, 1000, 102400, 1048576, 0.8}, // 3 nodes, 1000 keys, 100KB-1MB values, 80/20 read/writes
+	}
+
+	for _, bc := range benchmarksCases {
+		bName := fmt.Sprintf(
+			"N=%d,R=%d,K=%d,minBytes=%db,maxBytes=%db,wrRatio=%.2f",
+			bc.clusterSize,
+			bc.replicas,
+			bc.numKeys,
+			bc.minBytes,
+			bc.maxBytes,
+			bc.rwRatio,
+		)
+		b.Run(
+			bName,
+			func(b *testing.B) {
+				if verbose {
+					b.Logf("Setting up %d nodes", bc.clusterSize)
+				}
+				var connectURL string
+				if bc.clusterSize == 1 {
+					s := RunBasicJetStreamServer(b)
+					defer s.Shutdown()
+					connectURL = s.ClientURL()
+				} else {
+					cl := createJetStreamClusterExplicit(b, "BENCH_OBJ_STORE", bc.clusterSize)
+					defer cl.shutdown()
+					cl.waitOnClusterReadyWithNumPeers(bc.clusterSize)
+					cl.waitOnLeader()
+					connectURL = cl.randomServer().ClientURL()
+				}
+
+				nc, js := jsClientConnectURL(b, connectURL)
+				defer nc.Close()
+
+				// Pre-generate all keys
```

Comment:
Since your keys are K_{n}, and nor some random string, not much point in generating all of them ahead.

Rather than choosing a random key with `keys[rand(numKeys)]` you can just `sprintf("K_%d", rand(numKeys))`